cmd/edit/cluster/cmd.go

```diff
@@ -545,6 +550,39 @@ func run(cmd *cobra.Command, _ []string) {
 	r.Reporter.Infof("Updated cluster '%s'", clusterKey)
 }
 
+// warnUserForOAuthHCPVisibility is a method for HCP only that checks if the user has public ingress and warns them
+// about how changing cluster visibility may impact them
+func warnUserForOAuthHCPVisibility(r *rosa.Runtime, clusterKey string, cluster *cmv1.Cluster,
+	privateWarning string) (string, error) {
+	if !cluster.Hypershift().Enabled() {
+		return privateWarning, nil
+	}
+	// if ingress visibility public, warning
+	r.Reporter.Debugf("Loading ingresses for cluster '%s'", clusterKey)
+	ingresses, err := r.OCMClient.GetIngresses(cluster.ID())
```

Comment:
Note that only one ingress is supported for HCP clusters.
So, we can skip the query if you like (having more ingresses are not on the radar).